content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md

```diff
@@ -0,0 +1,564 @@
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+
+1. You must have a good understanding of Python
+2. You must have an excellent working knowledge of machine learning models
+3. You must have docker installed in your machine.
+4. You must have Kubernetes installed in your machine.
+
+### Building the machine learning model
+
+We will build a machine learning model that will predict the nationality of individuals using their names.
+This is a simple model that will explain the key concepts used in machine learning modeling.
+
+### Dataset to be used
+
+The dataset used will contains common names of people and their nationalities.
+Our data used is as shown.
+
+![A Snip of the data](/engineering-education/how-to-create-a-machine-learning-app-using-the-fast API-and-deploying-it-to-Kubernetes-cluster/snip.png)
+[CSV File of data](https://drive.google.com/file/d/1UNIqFDaqmfRxPmQaOC8SCPOQln27tfr-/view?usp=sharing)
+
+### Installation of the python packages
+
+We will use the following packages when building our model.
+
+1. [Pandas](https://pandas.pydata.org/)
+2. [Numpy](https://numpy.org/)
+3. [Sckit-learn](https://scikit-learn.org/)
+
+Run the following commands to install the packages.
+
+```bash
+pip install pandas
+pip install NumPy
+pip install sklearn
+```
+
+### Loading our EDA packages
+
+These packages are used for Exploratory data analysis to summarise the main characteristics of our data for easy visualization.
+
+```python
+import pandas as pd
+import NumPy as np
+```
+
+### Loading from Scikit-learn package
+
+[Scikit-learn](https://scikit-learn.org/) will be the package used for predictive analysis since it contains different tools for machine learning modeling and various algorithms for classification, regression, and clustering.
+
+```python
+import MultinomialNB from sklearn.naive_bayes
+import CountVectorizer from sklearn.feature_extraction.text
+import train_test_split from sklearn.model_selection
+import accuracy_score from sklearn.metrics
+```
+
+We will use the [Naive Bayes Classifier](https://en.wikipedia.org/wiki/Naive_Bayes_classifier) for our modeling.
+
+### Loading our dataset
+
+```python
+df = pd.read_csv("nationality.csv")
+```
+
+### Nature of our data
+
+We need to understand the nature of the dataset that we have. For example, we need to know the number of names in the dataset, the columns, and the rows present in the data.
+
+```python
+df.shape
+```
+
+```python
+df.head
+```
+
+```python
+df.columns
+```
+
+### All the nationalities available in our data
+
+```python
+df['nationality'].unique()
+```
+
+### Checking if our data is balanced
+
+```python
+df.groupby('nationality')['names'].size()
+```
+
+### Visualizing our data using the matplotlib library
+
+[Matplotlib](https://matplotlib.org/) is a Python library used for plotting hence easy visualization of our data in the form of a graph.
+
+```python
+import matplotlib.pyplot as plt
+%matplotlib inline
```

Comment:
If you are using magic commands, then do the following:
1. Initially, before beginning the explanation, inform the user that we will be working with Jupyter Notebooks.
2. Paste a link to Colab Notebook so that the reader can verify the results easily.